sympy/solvers/tests/test_recurr.py

```diff
@@ -184,6 +184,10 @@ def test_rsolve():
             (C0*((sqrt(-a**2 + 1) - 1)/a)**n +
              C1*((-sqrt(-a**2 + 1) - 1)/a)**n)).simplify() == 0
 
+    assert rsolve((k + 1)*y(k), y(k)) is None
+    assert (rsolve((k + 1)*y(k) + (k + 3)*y(k + 1) + (k + 5)*y(k + 2), y(k))
+            is None)
```

Comment:
Well, I did this due to the failures in computing ``FormalPowerSeries``. The algorithm converts a function into a RE eventually. That I try to solve using a function. If it's unsuccessful I fallback to more general ``rsolve``. 

I really don't like the failures.

The first RE is what you get when the function is ``1/x``. I will like it to fail silently.
In this particular case, I can check if there is a single term, don't compute.

The second RE is computed when the function is ``(x**2 + x + 1)/x**3``. Now this kind of case is hard to capture. As, many RE of this form are solved RE.

One way is to return ``None`` like I did. Otherwise we can also raise an explicit exception.  In general, ``rsolve`` returns ``None`` whenever it's not able to compute a valid RE.

What do you think?